packages/node_modules/@webex/plugin-meetings/src/meeting/audio.js

```diff
@@ -0,0 +1,99 @@
+import StateMachine from 'javascript-state-machine';
+import StateMachineHistory from 'javascript-state-machine/lib/history';
+
+import Media from '../media';
+import MeetingUtil from '../meeting/util';
+import {MEETING_AUDIO_STATE_MACHINE, EVENT_TRIGGERS, EVENT_TYPES} from '../constants';
+
+const handleTransition = (value, audio, meeting) => {
+  if (audio.mute && audio.self) {
+    value = MEETING_AUDIO_STATE_MACHINE.STATES.MUTE_SELF;
+  }
+  if (!audio.mute && audio.self) {
+    value = MEETING_AUDIO_STATE_MACHINE.STATES.UNMUTE_SELF;
+  }
+  return value;
+};
+
+const AudioStateMachine = {
+  create() {
+    return new StateMachine({
+      transitions: [
+        {
+          name: MEETING_AUDIO_STATE_MACHINE.TRANSITIONS.INIT,
+          from: '*',
+          to: function (audio, meeting) {
+            return handleTransition(this.state, audio, meeting);
```

Comment:
We don't actually use this.state (value) in the method above. Pedantic, but it makes more sense to do `this.state = handleTransition(audio, meeting)` and then `return this.state`; We're just overwriting the value in the helper method which isn't intuitive when looking at the method.

Reply:
following state machine standard here where the to function returns what value should be set because this.state gets automatically set from the to: function